src/main/java/uk/gov/justice/hmpps/prison/service/CaseNoteService.java

```diff
@@ -236,13 +238,40 @@ public List<CaseNoteUsageByBookingId> getCaseNoteUsageByBookingId(final String t
         return caseNoteRepository.getCaseNoteUsageByBookingId(type, subType, bookingIds, deriveDates.getFromDateToUse(), deriveDates.getToDateToUse());
     }
 
-    public List<CaseNoteUsageByBookingId> getCaseNoteUsageByBookingIdTypeAndDate(@NotEmpty final List<String> types, @NotNull final List<BookingFromDatePair> bookingReviewDatePairs) {
-        return bookingReviewDatePairs.stream()
-            .map(b -> caseNoteRepository.getCaseNoteUsageByBookingIdAndFromDate(types, b.getBookingId(), b.getFromDate()))
-            .flatMap(List::stream)
-            .collect(Collectors.toList());
+    public List<CaseNoteUsageByBookingId> getCaseNoteUsageByBookingIdTypeAndDate(@NotEmpty final List<String> types, @NotEmpty final List<BookingFromDatePair> bookingReviewDatePairs) {
+        final var bookingDateMap = bookingReviewDatePairs.stream().collect(Collectors.toMap(BookingFromDatePair::getBookingId, BookingFromDatePair::getFromDate));
+
+        final var allCaseNotesOfType = offenderCaseNoteRepository.findByOffenderBooking_BookingIdInAndType_CodeInAndOccurrenceDateTimeGreaterThanEqual(
+            bookingDateMap.keySet().stream().toList(),
+            types,
+            bookingDateMap.values().stream().min(LocalDateTime::compareTo).orElseThrow()
+        );
+
+        return allCaseNotesOfType.stream()
+            .filter(b -> !bookingDateMap.get(b.getOffenderBooking().getBookingId()).isAfter(b.getOccurrenceDateTime()))
+            .collect(groupingBy(cn -> new CaseNoteTypesAndSubTypes(cn.getOffenderBooking().getBookingId(), cn.getType().getCode(), cn.getSubType().getCode())))
+            .entrySet().stream()
+            .map(summary ->
+            {
+                final var key = summary.getKey();
+                final var value = summary.getValue();
+                return new CaseNoteUsageByBookingId(
+                    key.bookingId,
+                    key.type,
+                    key.subType,
+                    value.size(),
+                    value.stream().max(Comparator.comparing(OffenderCaseNote::getOccurrenceDateTime)).orElseThrow().getOccurrenceDateTime()
```

Comment:
doesn't look like this field is used in incentives - https://github.com/ministryofjustice/hmpps-incentives-api/blob/ad748b2342caf47643f94deced8f82ae6392b0fb/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt#L75

Reply:
Its not - This DTO is overloaded so I could create a new one without the field